app/libs/classes/viva_my_pages.py

```diff
@@ -14,63 +14,72 @@ def __init__(self, wsdl='MyPages', user=str):
         self._user = user
         self._pnr = self._user
 
-        self.person_info = self._get_person_info()
         self.person_cases = self._get_person_cases()
-        self.person_booked_payments = self._get_person_booked_payments()
-        self.person_caseworkflow = self._get_person_caseworkflow()
-        self.person_application = self._get_person_application()
 
-    def get_workflow(self, workflow_id=str):
-        if not workflow_id:
-            raise Fault(message='workflow_id is missing', code=400)
+    def get_workflow(self, workflow_id=None):
+        assert isinstance(
+            workflow_id, str), f'workflow_id should be type str. Got {type(workflow_id)}'
 
-        if not self.person_caseworkflow['vivadata']['vivacaseworkflows']['workflow']:
+        person_caseworkflow = self._get_person_caseworkflow(limit=6)
+
+        if not person_caseworkflow['vivadata']['vivacaseworkflows']['workflow']:
             raise Fault(message='No workflows found', code=404)
 
-        workflows = self.person_caseworkflow['vivadata']['vivacaseworkflows']['workflow']
+        workflow_list = person_caseworkflow['vivadata']['vivacaseworkflows']['workflow']
 
-        if not type(workflows) is list:
-            workflows = [workflows]
+        if not isinstance(workflow_list, list):
+            workflow_list = [workflow_list]
 
         workflow = next(
-            (item for item in workflows if item['workflowid'] == workflow_id), None)
+            (item for item in workflow_list if item['workflowid'] == workflow_id), None)
 
         if not workflow:
             raise Fault(
                 message=f'workflow with id: {workflow_id} not found', code=404)
 
         return workflow
 
-    def get_phone_number(self):
-        viva_case = self.person_cases['vivadata']['vivacases']['vivacase']
+    def get_workflow_list(self):
+        person_caseworkflow = self._get_person_caseworkflow(limit=6)
+
+        if not person_caseworkflow['vivadata']['vivacaseworkflows']['workflow']:
```

Comment:
I think this check could be made in a better way to avoid missing keys. Maybe something along the lines of def 

```
keys_exists(element: dict, *keys):
    if not isinstance(element, dict):
        ... raise error if not dict
    if len(keys) == 0:
        ...raise error if not all key kwargs does not exist
     
     # copy the element
    _element = element
    
    # check if keys exists in element
    for key in keys:
        try:
            _element = _element[key]
        except KeyError:
            return False
    return True
 
 # Usage
 test_dict = {
    'a':  {
          'b':   {
              'c': 'value'
          }
    }
 }
 
 keys_exsists(test_dict, 'a', 'b', 'c')
    ```

Reply:
Is this necessary? accessing keys that does not exist will throw a `KeyError` regardless. The entire method could thus be refactored into:

```python
  def get_workflow_list(self):
      person_caseworkflow = self._get_person_caseworkflow(limit=6)

      try:
          return person_caseworkflow['vivadata']['vivacaseworkflows']['workflow']
      except KeyError:
          raise Fault(message='No workflows found', code=404)
```

Which I think should be more performant as well.

Reply:
Agree with @KenthLjungHBG